pinot-common/src/main/java/org/apache/pinot/common/metrics/BrokerMetrics.java

```diff
@@ -84,4 +91,23 @@ protected BrokerMeter[] getMeters() {
   protected BrokerGauge[] getGauges() {
     return BrokerGauge.values();
   }
+
+  // Emits metrics based on BrokerResponse. Currently only emits metrics for exceptions.
+  // If a broker response has multiple exceptions, we will emit metrics for all of them.
+  // Thus, the sum total of all exceptions is >= total number of queries impacted.
+  // Additionally, some parts of code might already be emitting metrics for individual error codes.
+  // But that list isn't accurate with a many-to-many relationship (or no metrics) between error codes and metrics.
+  // This method ensures we emit metrics for all queries that have exceptions with a one-to-one mapping.
+  public void emitBrokerResponseMetrics(BrokerResponse brokerResponse) {
```

Comment:
Let me know if this method should belong somewhere else

Reply:
I think it is fine here, but please change the simple comment (`//`) to one of the javadoc formats (either the new `///` or the good old classic `/**/)

Reply:
yea that was the other option. moved to BrokerResponse